### ISSUE TITLE
Failed to load login after clean install via PPA

### ISSUE THREAD

**Issue by arcanisgk** [phpmyadmin/phpmyadmin#18597]
SO: Ubuntu 22.04
Version installed of phpmyadmin: 5.2.1

Command to reproduce it:

Mysql Command:
```
sudo apt install mysql-server -y --fix-missing
sudo mysql_secure_installation
# root password "*@q12345"
# disabled root remotely
# no plugins
# removed test database
# flushed priv
#post installation command on mysql query:
mysql> ALTER USER 'root'@'localhost' IDENTIFIED WITH mysql_native_password BY '*@q12345';
#edit mysql access:
sudo nano /etc/mysql/mysql.conf.d/mysqld.cnf
#comment this line: #bind-address = 127.0.0.1
```

Apache2 default installation: 
```sudo apt install -y --fix-missing apache2```

PHP command installation:
```
sudo add-apt-repository ppa:ondrej/php
sudo apt update && sudo apt upgrade -y --fix-missing
sudo DEBIAN_FRONTEND=noninteractive apt -y --fix-missing install php8.1 libapache2-mod-php8.1 php8.1-amqp php8.1-ast php8.1-bcmath php8.1-bz2 php8.1-calendar php8.1-cgi php8.1-cli php8.1-common php8.1-curl php8.1-dba php8.1-decimal php8.1-dev php8.1-ds php8.1-enchant php8.1-excimer php8.1-facedetect php8.1-fpm php8.1-gd php8.1-gearman php8.1-gmp php8.1-gnupg php8.1-grpc php8.1-http php8.1-igbinary php8.1-imagick php8.1-imap php8.1-inotify php8.1-interbase php8.1-intl php8.1-ldap php8.1-libvirt-php php8.1-lz4 php8.1-mailparse php8.1-maxminddb php8.1-mbstring php8.1-mcrypt php8.1-memcache php8.1-memcached php8.1-mongodb php8.1-msgpack php8.1-mysql php8.1-oauth php8.1-odbc php8.1-opcache php8.1-pcov php8.1-pgsql php8.1-phpdbg php8.1-pinba php8.1-protobuf php8.1-ps php8.1-pspell php8.1-psr php8.1-raphf php8.1-rdkafka php8.1-readline php8.1-redis php8.1-rrd php8.1-smbclient php8.1-snmp php8.1-soap php8.1-solr php8.1-sqlite3 php8.1-ssh2 php8.1-stomp php8.1-swoole php8.1-sybase php8.1-tideways php8.1-tidy php8.1-uopz php8.1-uploadprogress php8.1-uuid php8.1-vips php8.1-xdebug php8.1-xhprof php8.1-xml php8.1-xmlrpc php8.1-xsl php8.1-yaml php8.1-zip php8.1-zmq php8.1-zstd
```

phpMyAdmin command Installation:
```
sudo add-apt-repository ppa:phpmyadmin/ppa
sudo apt-get update && sudo apt -y --fix-missing install phpmyadmin
# put yes to all and paswword the same that mysql: *@q12345
```

Up to this point I assume that phpMyAdmin should already be functional... I try to enter to see the result of the installation and I get this, Error related:

![image](https://github.com/phpmyadmin/phpmyadmin/assets/1207306/358f2162-d388-4bf7-8ac3-86e5f31fc854)

Other try to install  5.1.1 version of phpMyAdmin:
```
sudo apt install -y --fix-missing phpmyadmin=4:5.1.1+dfsg1-5ubuntu1
```

not work, error becouse dependencies.

I'll be updating this post very soon as I'll be doing some testing before trying the manual install to see if this fixes the error.

<hr>

### UPDATE

Before doing the manual installation, I decided to update components and packages, run the command:

```
sudo apt update && sudo apt full-upgrade -y --fix-missing
```

The error is still not clear to me, but well, I assume that there was some package or dependency pending installation and the main errors in the php code were corrected, but I still get the following.

![image](https://github.com/phpmyadmin/phpmyadmin/assets/1207306/76c1d86d-2bda-4261-bd74-6e2234281888)

When I clean the browser's cache and go to http://localhost/phpmyadmin, certain errors appear that seem more gloomy and tragic to me:

![image](https://github.com/phpmyadmin/phpmyadmin/assets/1207306/a8ecbbca-c7e8-4ccb-9725-7d6cee3695f9)

**Comment by arcanisgk** [comment]
I spent all afternoon thinking about the issue and it seems that there is no way to solve it using the PPA.
so i did my research related to package compatibility. according to the phpmyadmin documentation:

Related to this POST: https://github.com/phpmyadmin/phpmyadmin/wiki/DebianUbuntu
Related to this seccion: https://github.com/phpmyadmin/phpmyadmin/wiki/DebianUbuntu#with-add-apt-repository-you-should-be-using-a-lts-ubuntu-version-

Not work on Ubuntu 22.04!!!
Related to this:

![image](https://github.com/phpmyadmin/phpmyadmin/assets/1207306/52287b33-707a-43a8-a995-91e53fb1075f)

I must think that the version that should be installed by default is 5.1.1, which does not happen, when verifying the installed version it is 5.2.1 forced by the same system when trying to search for version 5.1.1 It exists but it is impossible to perform the installation because the dependencies cannot be found:

```
sudo apt install -y --fix-missing phpmyadmin=4:5.1.1+dfsg1-5ubuntu1
```
can't install!!!

<hr>

### Solution:

Install php 7.4 and set default:
```
sudo DEBIAN_FRONTEND=noninteractive apt -y --fix-missing install php7.4 libapache2-mod-php7.4 libapache2-mod-php7.4 php7.4-amqp php7.4-ast php7.4-bcmath php7.4-bz2 php7.4-calendar php7.4-cgi php7.4-cli php7.4-common php7.4-ctype php7.4-curl php7.4-dba php7.4-decimal php7.4-dev php7.4-dom php7.4-ds php7.4-enchant php7.4-excimer php7.4-exif php7.4-facedetect php7.4-fileinfo php7.4-fpm php7.4-ftp php7.4-gd php7.4-gearman php7.4-geoip php7.4-gettext php7.4-gmp php7.4-gnupg php7.4-grpc php7.4-http php7.4-iconv php7.4-igbinary php7.4-imagick php7.4-imap php7.4-inotify php7.4-interbase php7.4-intl php7.4-json php7.4-ldap php7.4-libvirt-php php7.4-lua php7.4-lz4 php7.4-mailparse php7.4-maxminddb php7.4-mbstring php7.4-mcrypt php7.4-memcache php7.4-memcached php7.4-mongodb php7.4-msgpack php7.4-mysql php7.4-mysqli php7.4-mysqlnd php7.4-oauth php7.4-odbc php7.4-opcache php7.4-pcov php7.4-pdo php7.4-pdo-dblib php7.4-pdo-firebird php7.4-pdo-mysql php7.4-pdo-odbc php7.4-pdo-pgsql php7.4-pdo-sqlite php7.4-pgsql php7.4-phar php7.4-phpdbg php7.4-pinba php7.4-posix php7.4-propro php7.4-protobuf php7.4-ps php7.4-pspell php7.4-psr php7.4-radius php7.4-raphf php7.4-rdkafka php7.4-readline php7.4-redis php7.4-rrd php7.4-shmop php7.4-simplexml php7.4-smbclient php7.4-snmp php7.4-soap php7.4-sockets php7.4-solr php7.4-sqlite3 php7.4-ssh2 php7.4-stomp php7.4-swoole php7.4-sybase php7.4-sysvmsg php7.4-sysvsem php7.4-tideways php7.4-tidy php7.4-tokenizer php7.4-uopz php7.4-uploadprogress php7.4-uuid php7.4-vips php7.4-xdebug php7.4-xhprof php7.4-xml php7.4-xmlreader php7.4-xmlrpc php7.4-xmlwriter php7.4-xsl php7.4-yaml php7.4-zip php7.4-zmq php7.4-zstd
```

```
sudo update-alternatives --set php /usr/bin/php7.4
sudo a2dismod php8.1
sudo a2enmod php7.4
sudo apt install phpmyadmin -y --fix-missing
```
edit and update phpmyadmin configuration for apache

```
cd /etc/apache2/conf-available
sudo nano phpmyadmin.conf
```

comment php8+:

```
# phpMyAdmin default Apache configuration

Alias /phpmyadmin /usr/share/phpmyadmin

<Directory /usr/share/phpmyadmin>
    Options SymLinksIfOwnerMatch
    DirectoryIndex index.php

    # limit libapache2-mod-php to files and directories necessary by pma
    <IfModule mod_php7.c>
        php_admin_value upload_tmp_dir /var/lib/phpmyadmin/tmp
        php_admin_value open_basedir /usr/share/phpmyadmin/:/usr/share/doc/phpmyadmin/:/etc/phpmyadmin/:/var/lib/phpmyadmin/:/usr/share/php/:/usr/share/javascript/
    </IfModule>

    # PHP 8+
    #<IfModule mod_php.c>
    #    php_admin_value upload_tmp_dir /var/lib/phpmyadmin/tmp
    #    php_admin_value open_basedir /usr/share/phpmyadmin/:/usr/share/doc/phpmyadmin/:/etc/phpmyadmin/:/var/lib/phpmyadmin/:/usr/share/php/:/usr/share/javascript/
    #</IfModule>

</Directory>

# Disallow web access to directories that don't need it
<Directory /usr/share/phpmyadmin/templates>
    Require all denied
</Directory>
<Directory /usr/share/phpmyadmin/libraries>
    Require all denied
</Directory>
```

Manual phpMyAdmin installation:
```
apt -y install tar
sudo mv /usr/share/phpmyadmin/ /usr/share/phpmyadmin.bak
sudo mkdir /usr/share/phpmyadmin/
cd /usr/share/phpmyadmin/
sudo wget https://files.phpmyadmin.net/phpMyAdmin/5.2.1/phpMyAdmin-5.2.1-all-languages.tar.gz
sudo tar xzf phpMyAdmin-5.2.1-all-languages.tar.gz
sudo mv phpMyAdmin-5.2.1-all-languages/* /usr/share/phpmyadmin
sudo cp config.sample.inc.php config.inc.php
sudo cp config.sample.inc.php config.inc.php
sudo find /usr/share/phpmyadmin/ -type d -exec chmod 777 {} ";"
sudo find /usr/share/phpmyadmin/ -type f -exec chmod 777 {} ";"
sudo chmod 0755 config.inc.php
sudo nano config.inc.php                           #add blowfish_secret
sudo reboot
```


## In the end it seems that there is only one problem with the phpmyadmin package that there is no easy and guaranteed way to install through the PPA

![image](https://github.com/phpmyadmin/phpmyadmin/assets/1207306/a183426e-c54a-48a4-9107-224c22cf9fb8)

**Comment by williamdes** [comment]
TLDR for today, but this caught my attention:

> I must think that the version that should be installed by default is 5.1.1, which does not happen, when verifying the installed version it is 5.2.1 forced by the same system when trying to search for version 5.1.1 It exists but it is impossible to perform the installation because the dependencies cannot be found:
sudo apt install -y --fix-missing phpmyadmin=4:5.1.1+dfsg1-5ubuntu1

What dependencies ?
Can you past the output ?
What does `apt policy phpmyadmin` say ?

**Comment by arcanisgk** [comment]
> TLDR for today, but this caught my attention:
> 
> > I must think that the version that should be installed by default is 5.1.1, which does not happen, when verifying the installed version it is 5.2.1 forced by the same system when trying to search for version 5.1.1 It exists but it is impossible to perform the installation because the dependencies cannot be found:
> sudo apt install -y --fix-missing phpmyadmin=4:5.1.1+dfsg1-5ubuntu1
> 
> What dependencies ?
> Can you past the output ?
> What does `apt policy phpmyadmin` say ?


command:

```
sudo apt install phpmyadmin=4:5.1.1+dfsg1-5ubuntu1 -y --fix-missing
```

Output:

```
Reading package lists... 
Done Building dependency tree... 
Done Reading state information... 
Done Some packages could not be installed. 
This may mean that you have requested an impossible situation or if you are using the unstable distribution that some required packages have not yet been created or been moved out of Incoming. 
The following information may help to resolve the situation: 
The following packages have unmet dependencies: phpmyadmin : 
Depends: php-phpmyadmin-shapefile (< 3~~) but 3.0.1-2+jammy1 is to be installed Depends:
php-twig-i18n-extension (< 4~~) but 4.0.1-2+jammy1 is to be installed Recommends: php-tcpdf but it is not going to be installed E: Unable to correct problems, you have held broken packages.
```

**Comment by williamdes** [comment]
Hi @arcanisgk 

Sorry, this got out of my radar
Is this installation issue still existing ?
If yes, I will push the needed packages to the PPA shortly :)

**Comment by arcanisgk** [comment]
Hello friend, I haven't really tried it again.
If you haven't made changes related to this, it may still happen.

**Comment by williamdes** [comment]
> @williamdes hey I hope you'll excuse me, lately I've been giving them more work than usual. when I'm out of work I get idle with libraries and tools.

Thank you for posting about this install issue, I will update the version on the PPA very soon

**Comment by williamdes** [comment]
Ubuntu Noble PPA is up to date: https://github.com/phpmyadmin/phpmyadmin/issues/19589
Jammy is still in progress
